app/dashboard/src/components/AriaComponents/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,290 @@
+      invariant(value != null, 'Checkbox must have a value when inside a group')
+
+      return {
+        isSelected: state.selected.has(value),
+        // This is safe, because the name is handled by the `CheckboxGroup` component
+        // and checked there
+        // eslint-disable-next-line no-restricted-syntax
+        field: state.field as UseFormRegisterReturn<Schema, TFieldValues, TFieldName>,
+        // eslint-disable-next-line no-restricted-syntax
+        name: state.name as TFieldName,
+        onChange: (checked: boolean) => {
+          if (checked) {
+            addSelected(value)
+          } else {
+            removeSelected(value)
+          }
+        },
+      }
+    } else {
+      invariant(props.name != null, 'Checkbox must have a name when outside a group')
+
+      // eslint-disable-next-line no-restricted-syntax
+      const fieldInstance = formInstance.register(props.name) as UseFormRegisterReturn<
+        Schema,
+        TFieldValues,
+        TFieldName
+      >
+
+      return {
+        field: fieldInstance,
+        name: props.name,
+        isSelected: props.isSelected ?? false,
+        onChange: (checked: boolean) => {
+          void fieldInstance.onChange({ target: { value: checked } })
+          void formInstance.trigger(props.name)
+        },
+      }
+    }
+  })
+
+  const { fieldState } = Form.useField({
+    name,
+    // eslint-disable-next-line no-restricted-syntax
+    form: formInstance as unknown as Parameters<typeof Form.useField>[0]['form'],
+  })
+
+  const classes = variants({
+    isReadOnly: isReadOnly,
+    isInvalid: isInvalid || fieldState.invalid,
+    isDisabled: isDisabled || field.isDisabled,
+    size,
+  })
+
+  return (
+    <AriaCheckbox
+      ref={mergeRefs(ref, field.ref)}
+      {...props}
+      /* eslint-disable-next-line no-restricted-syntax */
+      inputRef={checkboxRef as MutableRefObject<HTMLInputElement>}
+      className={(renderProps) => classes.base({ className, isSelected: renderProps.isSelected })}
+      isSelected={isSelected}
+      onChange={onChange}
+      onBlur={field.onBlur}
+      isIndeterminate={isIndeterminate}
+      isInvalid={isInvalid || fieldState.invalid}
+      isDisabled={isDisabled || (field.isDisabled ?? false)}
+      isReadOnly={isReadOnly}
+      isRequired={field.required ?? false}
+    >
+      {(renderProps) => (
+        <>
+          <motion.svg
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="0 0 16 16"
+            className={classes.icon({ isSelected: renderProps.isSelected })}
+            initial={false}
+            animate={renderProps.isSelected ? 'checked' : 'unchecked'}
+            role="presentation"
+          >
+            <motion.path
+              strokeLinecap="round"
+              strokeLinejoin="round"
+              strokeWidth="2"
+              stroke="currentColor"
+              fill="none"
+              d={isIndeterminate ? 'M5 8H11' : 'M4 8.4L6.5 10.9L9.25 8.15L12 5.4'}
```

Comment:
i think these would be more descriptive as constants.

also personal preference but i prefer relative (lowercase) draw commands so they can be moved around easily without hacks like transforms